frontend/src/wordInteractions.js

```diff
@@ -53,21 +82,64 @@ function playerTypesWord () {
   typingForm.addEventListener('submit', (event) => {
     event.preventDefault();
     const typedSubmission = typingForm['player-input'].value;
-    WORDS_TYPED.push(typedSubmission);
-    if (WORDS_SEEN.includes(typedSubmission)) {
-      removeWordFromPage(typedSubmission);
+    const matchOnPage = document.getElementById(typedSubmission);
+
+    if (removeWordFromPage(typedSubmission)) {
+      game.wordsTyped.push(typedSubmission);
     } else {
-      console.log('no match on page');
+      game.typos++;
+    }
+    if (game.typos > 2) {
+      gameOver(false);
+    } else if (allWordsTyped()) {
+      gameOver(true);
     }
+
     typingForm['word-entered'].value = '';
   });
 }
 
+function allWordsTyped () {
+  return game.wordsTyped.length === game.wordsSeen.length;
```

Comment:
This feels buggy. Someone could type really fast and clear all of the words on the screen, which would trigger the end of the game. I think you actually only want this to trigger if all the words in our database have been typed (ALL_WORDS)

Reply:
ahh the point right now was that I wanted it to be winnable so I could test things, but yes later on we would remove this since there is no way to "win" the game.

Reply:
gotch. That's fine for now then